ui/src/components/Player/DetectionDialog.tsx

```diff
@@ -40,7 +39,7 @@ export default function DetectionDialog({
 }) {
   const [open, setOpen] = useState(false);
   const [submitted, setSubmitted] = useState(false);
-  const [category, setCategory] = useState<DetectionCategory>();
+  const [category, setCategory] = useState<DetectionCategory | undefined>();
```

Comment:
```suggestion
  const [category, setCategory] = useState<DetectionCategory>();
```
Type inference will pick up undefined, no need to add it